apache/incubator-kie-tools
# kie-issues#2091: Subflow support on CLI

Fixes: https://github.com/apache/incubator-kie-tools/issues/2091

We should enhance the current CLI to support subflows. To achieve that, we need to introduce a new workflow project default structure:

Workflow project root
/specs (optional)
/schemas (optional)
/subflows (optional)
workflow.json|yaml (mandatory)

To test:
[subflow1.zip](https://github.com/apache/incubator-kie-tools/files/13640914/subflow1.zip)

This PR also introduce a batch of improvements the CLI:

-New workflow project structure

-Upgrade kogito-swf-dev mode to 1.44

-Enable users to create workflows with YAML format
To test: ./cli create --yaml-workflow

-Changed all flags from camelCase to dash notation

-Quarkus convert should copy all content 

-Custom directory to deploy manifests
To test: 
./cli deploy --custom-manifests-dir=/Users/ederign/temp/subflows/sanity/sample-project/manifests


What is still pending:

- Update sonata flow docs
- Manifests will become optional: https://github.com/apache/incubator-kie-kogito-serverless-operator/pull/320 and we need to update the CLI.
